src/material/core/_core.scss

```diff
@@ -16,20 +16,33 @@
   @include text-field.text-field-autosize();
   @include text-field.text-field-autofill();
 
-  @include focus-indicators.private-strong-focus-indicators-positioning();
-  @include _mdc-core();
-}
-
-// Mixin that renders all of the core MDC styles. Private mixin included with `mat-core`.
-@mixin _mdc-core() {
-  @include _mdc-strong-focus-indicators-positioning();
-}
-
-// Mixin that ensures focus indicator host elements are positioned so that the focus indicator
-// pseudo element within is positioned relative to the host. Private mixin included within
-// `_mat-mdc-core`.
-@mixin _mdc-strong-focus-indicators-positioning() {
+  .mat-focus-indicator,
   .mat-mdc-focus-indicator {
     position: relative;
+
+    &::before {
+      --mat-focus-indicator-border-width: var(--mat-focus-indicator-border-width-custom, 3px);
```

Comment:
We shouldn't be defining the variable here. The idea is that the structural styles define the slots for the variables, e.g. `border-width: var(--mat-focus-indicator-border-width, 3px)` and then the mixin overrides the defaults.

Reply:
Recall that there are component-specific styles that depend on the focus indicator width (i.e. in order to know how much to inset/offset the focus indicator w/ `calc`. If we don't use two tiers of CSS variables like we're using here, then we'll need to have something like this in the component's stylesheet:

```
.mat-mdc-outlined-button .mat-mdc-focus-indicator::before {
  margin: calc(-(var(--mat-focus-indicator-border-width, $default-focus-indicator-width) + 3px));
}
```

Maybe that's fine? IIRC the purpose of the two tiers of CSS variables is to instead allow us to do:

```
.mat-mdc-outlined-button .mat-mdc-focus-indicator::before {
  margin: calc(-(var(--mat-focus-indicator-border-width) + 3px));
}
```

and not need to specify the default fallback of `$default-focus-indicator-width`, because `--mat-focus-indicator-border-width` would always be defined. I think I'm fine with either approach though.

Reply:
Hmm, good point. Since we control the defaults, it's fine to have a Sass variable for it to use as a default in the calculation. My only note is that we should prefix the Sass variable names with `private-` so users don't end up depending upon them.